From 689bce65eff45e77426f5f98795b19ee30ab2b26 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 12 Jun 2017 16:27:14 +0800
Subject: [PATCH 1955/5242] MLK-15066-1: ASoC: imx-pcm-dma-v2: fix noise issue
 with pulseaudio

commit  e1d9b402302bbf4ef72c356924f642049768716e from
https://source.codeaurora.org/external/imx/linux-imx.git

Same as commit c55075170214 ("MLK-14582: ASoC: imx-pcm-rpmsg: fix
audio noise issue with pulseaudio"), need to add a constraint for
SNDRV_PCM_HW_PARAM_PERIODS, which make the period number integer.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-pcm-dma-v2.c |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-pcm-dma-v2.c b/sound/soc/fsl/imx-pcm-dma-v2.c
index 6a3a36e..27ce201 100644
--- a/sound/soc/fsl/imx-pcm-dma-v2.c
+++ b/sound/soc/fsl/imx-pcm-dma-v2.c
@@ -115,12 +115,23 @@ static int imx_pcm_open(struct snd_pcm_substream *substream)
 		chan = dma_request_slave_channel(rtd->cpu_dai->dev,
 						 dma_data->filter_data);
 		ret = snd_dmaengine_pcm_open(substream, chan);
+		if (ret)
+			return ret;
 	} else {
 		ret = snd_dmaengine_pcm_open_request_chan(substream,
 							  imx_dma_filter_fn,
 							  dma_data->filter_data);
+		if (ret)
+			return ret;
 	}
-	return ret;
+
+	ret = snd_pcm_hw_constraint_integer(substream->runtime,
+					    SNDRV_PCM_HW_PARAM_PERIODS);
+	if (ret < 0)
+		return ret;
+
+
+	return 0;
 }
 
 static int imx_pcm_mmap(struct snd_pcm_substream *substream,
-- 
1.7.9.5

